### PR TITLE
Enhance Logs E2E test and bump up MySQL library to 8.4

### PR DESCRIPTION
**Pull request by ektabj**
*Issue description:*

In this PR:
- Supports `remoteResourceType` and `remoteResourceIdentifier` filter when fetching from Logs.
- Migrating MySQL to 8.4.
- Updated trace template.

---

E2E test after updating trace template: 
- https://github.com/ektabj/aws-application-signals-test-framework/actions/runs/10793931428/job/29937029029
- https://github.com/ektabj/aws-application-signals-test-framework/actions/runs/10792988948/job/29933883537

---

E2E test failed run reproduced locally - https://github.com/ektabj/aws-application-signals-test-framework/actions/runs/10771640665/job/29867588349
E2E test run passed after changes: https://github.com/ektabj/aws-application-signals-test-framework/actions/runs/10777657675/job/29887245097

Issue description:

When we assert `RemoteResourceType` and `RemoteResourceIdentifier` attributes in [EMF logs](https://github.com/aws-observability/aws-application-signals-test-framework/blob/main/validator/src/main/resources/expected-data-template/java/eks/aws-sdk-call-log.mustache#L30-L33), we are using Logs filter: `RemoteService` and `RemoteOperation` filter to fetch logs. However, it's not guaranteed that EMF log with `RemoteService` and `RemoteOperation` attributes will have `RemoteResourceType` and `RemoteResourceIdentifier`. The logs fetched with this generic filter might get more logs than expected and only asserting the first log event in the result.

Description of changes:

This change enhances the log filter by appending  `&& ($.RemoteResourceType = %EXAMPLE%) && ($.RemoteResourceIdentifier = %EXAMPLE%)` to the filter if they present in the template to get more specific log events.

Log Filter doc: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html#regex-expressions

Log Filter used before this change (From E2E testing log):
```
2024-07-02T13:41:46.9400277Z 13:41:46.840 [main] INFO  com.amazon.aoc.validators.CWLogValidator - Filter Pattern for Log Search: { ($.Service = sample-application-eu-west-1-9762031914-112) && ($.Operation = "GET /aws-sdk-call") && ($.RemoteService = "AWS::S3") && ($.RemoteOperation = "GetBucketLocation") }
```
Log Filter used after this change (From E2E testing log):
```
2024-07-02T13:49:00.6162597Z 13:49:00.510 [main] INFO  com.amazon.aoc.validators.CWLogValidator - Filter Pattern for Log Search: { ($.Service = sample-application-eu-west-1-9762033464-113) && ($.Operation = "GET /aws-sdk-call") && ($.RemoteService = "AWS::S3") && ($.RemoteOperation = "GetBucketLocation") && ($.RemoteResourceType = %^AWS::S3::Bucket$%) && ($.RemoteResourceIdentifier = %^e2e-test-bucket-name-eu-west-1-9762033464-113$%) }
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: validator/src/main/resources/expected-data-template/java/eks/rds-mysql-trace.mustache [hunk 1]
```diff
@@ -50,6 +50,32 @@
...
+                            "database_type": "^mysql$"
+                        },
+                        "annotations": {
+                            "aws.remote.operation": "^UnknownRemoteOperation$",
+                            "aws.local.operation": "^GET /mysql$",
+                            "aws.remote.resource.type": "^DB::Connection$",
+                            "aws.remote.resource.identifier": "^{{remoteResourceIdentifier}}$",
+                            "aws.remote.service": "^mysql$",
+                            "aws.local.service": "^{{serviceName}}$",
+                            "aws.local.environment": "^eks:{{platformInfo}}/{{appNamespace}}$"
```
Comment (jerry-shao) [inline validator/src/main/resources/expected-data-template/java/eks/rds-mysql-trace.mustache:66]: We don't care much about these things, as long as we know the first span is for Create connection.

File: validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java [hunk 2]
```diff
@@ -139,6 +141,10 @@ private Map<String, Object> getActualLog(
       dependencyFilter = String.format("&& ($.RemoteService = \"%s\") && ($.RemoteOperation = \"%s\")", remoteService, remoteOperation);
     }
 
+    if (remoteResourceType != null && remoteResourceIdentifier != null) {
+      dependencyFilter += String.format(" && ($.RemoteResourceType = %%%s%%) && ($.RemoteResourceIdentifier = %%%s%%)", remoteResourceType, remoteResourceIdentifier);
```
Comment (jerry-shao) [inline validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java:145]: From the 1st revision, we agreed to not using regex here right?

File: validator/src/main/resources/expected-data-template/java/eks/rds-mysql-trace.mustache [hunk 3]
```diff
@@ -73,8 +79,7 @@
                                 "K8s.Namespace": "^{{appNamespace}}$",
                                 "PlatformType": "^AWS::EKS$",
                                 "aws.span.kind": "^CLIENT$",
-                                "db.sql.table": "^tables$",
-                                "db.operation": "^SELECT$"
```
Comment (jerry-shao) [inline validator/src/main/resources/expected-data-template/java/eks/rds-mysql-trace.mustache:77]: May I know why do we delete this?

File: validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java (hunk 2 above)
Comment (ektabj) [reply validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java:145]: Nope, if we'll remove the regex then we'll have to do changes in all templates for EC2, EKS & K8s.
So Mahad agreed to not change it .

File: validator/src/main/resources/expected-data-template/java/eks/rds-mysql-trace.mustache (hunk 3 above)
Comment (ektabj) [reply validator/src/main/resources/expected-data-template/java/eks/rds-mysql-trace.mustache:77]: I actually removed that as part of previous template, I can add it back.

Comment (ektabj) [reply validator/src/main/resources/expected-data-template/java/eks/rds-mysql-trace.mustache]: Updated the template.

**Comment by jerry-shao** [comment]
LGTM

**Review by majanjua-amzn — APPROVED** [review]

**Pull request merged**